membership-attribute-service/app/services/zuora/payment/ZuoraPaymentService.scala

```diff
@@ -113,59 +90,34 @@ class PaymentService(zuoraService: ZuoraSoapService, planMap: Map[ProductRatePla
       case _ => None
     }
 
-  private def billingSchedule(subId: Id, accountFuture: Future[Account], numberOfBills: Int)(implicit
+  private def getNextBill(subId: Id, account: Account, numberOfBills: Int)(implicit
       logPrefix: LogPrefix,
-  ): Future[Option[BillingSchedule]] = {
-    val finder: ProductRatePlanChargeId => Option[Benefit] = planMap.get
-    val adapter: Seq[Queries.PreviewInvoiceItem] => Option[BillingSchedule] = BillingSchedule.fromPreviewInvoiceItems(finder)
-    val scheduleFuture = zuoraService.previewInvoices(subId, numberOfBills).map(adapter)
-    for {
-      account <- accountFuture
-      scheduleOpt <- scheduleFuture
-    } yield {
-      scheduleOpt.map(_.withCreditBalanceApplied(account.creditBalance))
+  ): Future[Option[Bill]] =
```

Comment:
before this returned the whole list of bills (invoices) and now it only gets the one actually needed